src/generator/boardGenerator.java

```diff
@@ -0,0 +1,222 @@
+package generator;
+
+import java.util.Random;
+
+public class boardGenerator {
+    private int[][] board;
+    private int size;
+
+    public boardGenerator(int size) {
+        size--;
+        if (size < 3) {
+            this.size = 3;
+        } else {
+            this.size = size;
+        }
+        board = new int[size][size];
+        generate();
+        while (checkReq())// hehe, yeah, generate 'til...
+            generate();
+        while (!setValues())
+            resetBoard();
+        show();
+        System.out.println(toString());
+    }
+
+    //this generator could generate board with two fields that don't have routes to one another
+
+    private void generate() {
+        Random rand = new Random();
+        int n;
+        for (int i = 0; i != size; ++i) {
+            for (int j = 0; j != size; ++j) {
+                if ((n = rand.nextInt(2)) == 0)
+                    n = rand.nextInt(2);
+                board[i][j] = 0 - n;
```

Comment:
TODO: use something like final int or enum, now I'm not totally sure what means which value